### PR TITLE
🎉 Source Salesforce: Support oauth'ing into sandbox environment, switching to advanced_auth

### LINKED ISSUE #7722: Source Salesforce: Support oauth'ing into sandbox environment
## Tell us about the problem you're trying to solve
When oauth'ing a salesforce user into a sandbox environment, we need to send them to `test.salesforce.com` rather than `login.salesforce.com`. Our current server implementation of salesforce oauth always assumes they are not trying to use a sandbox environment and redirects them to `login.salesforce.com`. 

Unfortunately, many enterprise users are not allowed to use production environments for testing new tools. Therefore, they would not be able to adopt Airbyte Cloud without being able to access the sandbox environment. 

Ideally, we would wait for #6971 to be implemented and use the functionality offered there to implement this flow. However, since that could take some time, we need to come up with a way to allow new Airbyte Cloud customers to start 
Once #6971 is implemented, we need to look at the user's input config and determine which oauth configuration to use. 

## Describe the solution you’d like

This fix has the following components: 
1. Disable the `sandbox` option of the Salesforce connector and always assume we are using the production environment (remove it from the spec but not from the code, and allow additional properties = true. if the undocumented option is passed then the connector should use sandbox) 
2. Create a fork of the salesforce connector, `source-salesforce-sandbox` (similar to the `source-file-secure` fork of the `source-file` connector) which always assumes we are using the salesforce sandbox environment. By the end we should have two salesforce connectors, `source-salesforce` which always assumes NOT sandbox, and `source-salesforce-sandbox` which always assumes sandbox environment. Publish both connectors normally like we do with any connector. 
3. Create a new oauth flow in the server which can log the user in for the sandbox environment. For testing this flow, use the `Salesforce (Sandbox)` Lastpass credentials. The only difference between this flow and the current salesforce flow is that the current salesforce flow uses the `login.salesforce.com` endpoint whereas the new flow should use `test.salesforce.com`. That's it. Otherwise, it can use the same credentials. 
4. Once you've tested the flow, create a PR and merge as normal on salesforce. 
5. Create a ticket to revert 1 & 2 once #6971 is implemented. That is, once that ticket is implemented, we should go back to having a single salesforce connector with a sandbox toggle whose oauth flow is controlled based on the toggle.

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
When oauth'ing a salesforce user into a sandbox environment, we need to send them to `test.salesforce.com` rather than `login.salesforce.com` Our current server implementation of salesforce oauth always assumes they are not trying to use a sandbox environment and redirects them to `login.salesforce.com`.

Closes #7722 

## How
Switched the input format of the Salesforce connector to use the new oauth specification. See the advanced_auth section [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L309). This changes also include server side changes for Salesforce oauth flow. 

## Pre-merge Checklist
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### CODE REVIEW

**Review by eliziario — APPROVED** [review]

**Comment by sherifnada** [comment]
@vitaliizazmic is there something blocking this change?

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.json [hunk 1]
```diff
@@ -7,6 +7,12 @@
     "required": ["client_id", "client_secret", "refresh_token", "api_type"],
     "additionalProperties": false,
     "properties": {
+      "auth_type": {
+        "type": "string",
+        "const": "Client",
+        "enum": ["Client"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.json:13]: need only `const` not `enum` and `default`

**Comment by vitaliizazmic** [comment]
> is there something blocking this change?

@sherifnada  no, I was waiting for additional review. I will publish and merge it.

**Pull request merged**